tests/fsharp/Compiler/Utilities.fs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.UnitTests
+
+open System
+open System.Collections.Immutable
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.CSharp
+
+// This file mimics how Roslyn handles their compilation references for compilation testing
+
+[<RequireQualifiedAccess>]
+type TargetFramework =
+    | NetStandard20
+    | NetCoreApp30
+
+module private TestReferences =
+
+    [<RequireQualifiedAccess>]
+    module NetStandard20 =
+
+        let NetStandard = lazy AssemblyMetadata.CreateFromImage(TestResources.NetFX.netstandard20.netstandard).GetReference(display = "netstandard.dll (netstandard 2.0 ref)")
```

Comment:
Nits:

* Lower-case as per the style guide (since we're not using an existing style here)
* `open` the namespace(s) you're using here, since you're using them in lots of places

Reply:
Applied to the all additions here

Reply:
Making this lowercase feels a bit weird. I'd argue to keep the casing as is.

Reply:
What's weird about it? It's idiomatic F# to have lower-case values unless they're literals.

Reply:
When the value is used, I don't like the way it looks when it is camel cased. If we are going to be super strict about casing rules for modules, then I will make this a class and use members. I prefer not though.

Reply:
It would make zero sense for these to be members and that would be highly non-idiomatic code.

We should be strict about this since it's new code. The [style guide](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#naming-conventions) isn't arbitrary, and we shouldn't arbitrarily disagree with it in our code unless we're working with an existing style (e.g., upper-case functions in the Typechecker).

Reply:
I disagree with "zero sense" here. If I make these members, I can have the backing field hold the lazy instance and the member would evaluate it. The use would be cleaner.

But, I agree we should be strict with style, even if I don't like it. I'll just make this lowercase so we can move on.